src/celengine/stardb.cpp

```diff
@@ -77,6 +78,15 @@ constexpr inline AstroCatalog::IndexNumber TYC3_MAX   = 3u;     // from TYC2
 constexpr inline AstroCatalog::IndexNumber TDSC_TYC3_MAX            = 4u;
 constexpr inline AstroCatalog::IndexNumber TDSC_TYC3_MAX_RANGE_TYC1 = 2907u;
 
+// stars.dat file header and record sizes
+constexpr inline std::size_t FILE_HEADER_SIZE = FILE_IDENTIFIER.size() + sizeof(std::uint16_t) + sizeof(std::uint32_t);
+constexpr inline std::size_t FILE_RECORD_SIZE = sizeof(AstroCatalog::IndexNumber)
+                                              + 3 * sizeof(float)
+                                              + sizeof(std::int16_t) + sizeof(std::uint16_t);
```

Comment:
This looks too fragile. What if we introduce structures for this purpose?
```
struct __attribute__((__packed__)) StarDBHeader
{
    unsigned char id[8];
    std::uint16_t version;
    std::uint32_t counter;
};

// and

struct __attribute__((__packed__)) StarDBRecord
{
    AstroCatalog::IndexNumber indexNumber;
    float x, y, z;
    std::int16_t absMag;
    std::uint16_t spectralType;
};
```

Reply:
Done